File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

bool cmp(vector<int> &a, vector<int> & b){
    if(a[1] == b[1]) return a[0] < b[0]; // 진출 같으면 진입 빠른 순
    return a[1] < b[1]; // 진출 빠른 순
}
int solution(vector<vector<int>> routes) {
    sort(routes.begin(), routes.end(), cmp);
    int answer = 0;

    for(int i = 0, j = i + 1; i < routes.size(); i = j, j = i+1){
        while(j < routes.size() && routes[i][1]; >= routes[j][0]) j++;
        answer++;
    }
    return answer;
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
            
        Arrays.sort(routes, (r1, r2) -> r1[1] - r2[1]);
        
        int cam = -30005;
        
        for(int[] r : routes){
            if(cam < r[0]){
                cam = r[1];
                answer++;
            }
        }
        
        return answer;
    }
}
